src/core_plugins/kibana/public/visualize/wizard/wizard.js

```diff
@@ -77,8 +89,11 @@ module.controller('VisualizeWizardStep2', function ($route, $scope, timefilter,
     if (!pattern) return;
 
     if (addToDashMode) {
-      return `#/visualize/create?${DashboardConstants.ADD_VISUALIZATION_TO_DASHBOARD_MODE_PARAM}&type=${type}&indexPattern=${pattern}`;
+      return `#${VisualizeConstants.CREATE_URL}` +
```

Comment:
There is a lot of this going on, I don't want to comment on it everywhere, but I'd like to reiterate my opinion that dynamically creating urls with string concatenation is a recipe for hard to trace bugs. Ideally we would always use a bonafide formatter like `url.format()`